diesel/src/types/impls/integers.rs

```diff
@@ -10,7 +10,7 @@ impl<DB: Backend<RawValue=[u8]>> FromSql<types::SmallInt, DB> for i16 {
         let mut bytes = not_none!(bytes);
         debug_assert!(bytes.len() <= 2, "Received more than 2 bytes decoding i16. \
                       Was an Integer expression accidentally identified as SmallInt?");
-        debug_assert!(bytes.len() == 2, "Received fewer than 2 bytes decoding i16. \
+        debug_assert!(bytes.len() >= 2, "Received fewer than 2 bytes decoding i16. \
```

Comment:
It's technically more correct, but my reasoning was actually "this is less typing than `#[allow]`"